acceptance/tests/restart_host_run_puppet.rb

```diff
@@ -22,8 +22,14 @@
   step "restart each agent" do
     applicable_agents.each do |agent|
       agent.reboot
+      # BKR-812
+      sleep 1 if !agent.up?
       assert(agent.up?, "Agent #{agent} should be up after reboot")
-      on(agent, puppet('resource service pxp-agent ')) do |result|
+      opts = {
+        :max_retries => 10,
+        :retry_interval => 1,
+      }
+      retry_on(agent, puppet('resource service pxp-agent '), opts) do |result|
```

Comment:
This retry will not help for the case where pxp-agent service reports as being "ensure=>stopped" after startup (PCP-393). In that case, this retry_on will succeed on its first attempt, with the result being "ensure=>stopped".